### PR TITLE
Update dependencies for 2.13.0-RC1

### PR DESCRIPTION
**Pull request by kubukoz**
Apparently, there are some new ambiguities in scalacheck under 2.11:

```
[error] /home/travis/build/functional-streams-for-scala/fs2/core/shared/src/test/scala/fs2/ChunkGen.scala:233:10: ambiguous reference to overloaded definition,
[error] both method apply in object Cogen of type (f: (org.scalacheck.rng.Seed, List[A]) => org.scalacheck.rng.Seed)org.scalacheck.Cogen[List[A]]
[error] and  method apply in object Cogen of type (implicit ev: org.scalacheck.Cogen[List[A]])org.scalacheck.Cogen[List[A]]
[error] match expected type ?
[error]     Cogen[List[A]].contramap(_.toList)
[error]          ^
[error] /home/travis/build/functional-streams-for-scala/fs2/core/shared/src/test/scala/fs2/FreeCSpec.scala:36:88: ambiguous reference to overloaded definition,
[error] both method apply in object Cogen of type (f: (org.scalacheck.rng.Seed, Unit) => org.scalacheck.rng.Seed)org.scalacheck.Cogen[Unit]
[error] and  method apply in object Cogen of type (implicit ev: org.scalacheck.Cogen[Unit])org.scalacheck.Cogen[Unit]
[error] match expected type ?
[error]       f <- arbFunction1[Result[A], FreeC[F, A]](Arbitrary(freeGen[F, A](fDepth)), Cogen[Unit].contramap(_ => ())).arbitrary
[error]                                                                                        ^
[error] /home/travis/build/functional-streams-for-scala/fs2/core/shared/src/test/scala/fs2/TestUtil.scala:100:73: ambiguous reference to overloaded definition,
[error] both method apply in object Cogen of type (f: (org.scalacheck.rng.Seed, List[A]) => org.scalacheck.rng.Seed)org.scalacheck.Cogen[List[A]]
[error] and  method apply in object Cogen of type (implicit ev: org.scalacheck.Cogen[List[A]])org.scalacheck.Cogen[List[A]]
[error] match expected type ?
[error]     implicit def pureStreamCoGen[A: Cogen]: Cogen[PureStream[A]] = Cogen[List[A]].contramap[PureStream[A]](_.get.toList)
[error]                                                                         ^
[error] three errors found
```

### CODE REVIEW

**Comment by travisbrown** [comment]
If I remember correctly this is because of the way the SAM stuff works on 2.11. You could try removing `-Xexperimental` and see what else breaks, or you could just write out `implicitly[Cogen[List[A]]]`. (Or you could drop 2.11 support.)

File: build.sbt [hunk 1]
```diff
@@ -49,12 +49,12 @@ lazy val commonSettings = Seq(
...
-    "org.typelevel" %%% "cats-laws" % "1.6.0" % "test",
-    "org.typelevel" %%% "cats-effect" % "1.3.0",
-    "org.typelevel" %%% "cats-effect-laws" % "1.3.0" % "test",
-    "org.scala-lang.modules" %%% "scala-collection-compat" % "0.3.0",
-    "org.scalatest" %%% "scalatest" % "3.0.6" % "test"
+    "org.typelevel" %%% "cats-core" % "2.0.0-M1",
+    "org.typelevel" %%% "cats-laws" % "2.0.0-M1" % "test",
+    "org.typelevel" %%% "cats-effect" % "2.0.0-M1",
+    "org.typelevel" %%% "cats-effect-laws" % "2.0.0-M1" % "test",
+    "org.scala-lang.modules" %%% "scala-collection-compat" % "1.0.0",
```
Comment (mpilquist) [inline build.sbt]: I think 1.0.0 was a bad build and we're supposed to stick with 0.3.0 until 2.0.0. From the README for that project:
> Note: Please do not release any artifacts against version 1.0.0. Due to #195 we are planning on releasing 2.0.0 that is neither source nor binary compatible with 1.0.0.

Comment (kubukoz) [reply build.sbt]: Oh. I didn't realize that. For sure it makes sense to wait for 2.0.0.

Also, these changes should be binary compatible for fs2, right? It doesn't define any testing modules that'd depend on the problematic dependencies of cats. (or does it)

Comment (mpilquist) [reply build.sbt]: #1454 removed the scala-collection-compat dependency altogether. I agree with your reasoning on why the cats M1 build is okay.

**Comment by travisbrown** [comment]
@kubukoz Hope you don't mind but I included this commit in #1457 together with everything else needed for 2.13.0-RC1.

**Comment by mpilquist** [comment]
Included in #1457

**Comment by kubukoz** [comment]
That's awesome @travisbrown, thanks for picking it up

**Pull request closed without merging**